controllers/provisioning_controller.go

```diff
@@ -266,7 +266,7 @@ func (r *ProvisioningReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	}
 
 	// Determine the status of the deployment
-	deploymentState, err := provisioning.GetDeploymentState(r.KubeClient.AppsV1(), ComponentNamespace, baremetalConfig)
+	deploymentState, err := provisioning.GetDeploymentState(r.KubeClient.AppsV1(), info.Namespace)
```

Comment:
ComponentNamespace and info.Namespace are equal. But wouldn't it better to use info.Namespace instead constant value in reconciler's context?